File: Programming-Basics-With-JavaScript/Homeworks-And-Labs/05.While-Loop/02-exercise/06.cake.js
function cake(input) {

    const width = Number(input[0]);
    const length = Number(input[1]);
    // Бележка: Едно парче торта е с размер 1х1 см.
    const numberOfPieces = width * length;
    let index = 2;
    let current = input[index++];
    let counterPieces = 0;

    while (current !== "STOP") {
        let currentPieces = Number(current);
        counterPieces += currentPieces;
        if (counterPieces >= numberOfPieces) {
            break;
        }

        current = input[index++];
    }

    const difference = Math.abs(numberOfPieces - counterPieces);
    if (numberOfPieces > counterPieces) {
        console.log(`${difference} pieces are left.`);
    } else {
        console.log(`No more cake left! You need ${difference} pieces more.`);
    }
}
cake(["10",
"2",
"2",
"4",
"6",
"STOP"]);

File: Python_Basics/Homeworks-And-Labs/05.while_loop/02.exercise/06.cake.py
width = int(input())
length = int(input())

#  Бележка: Едно парче торта е с размер 1х1 см.
number_of_pieces = width * length
current = input()
counter_pieces = 0

while current != "STOP":
    current_pieces = int(current)
    counter_pieces += current_pieces
    if counter_pieces >= number_of_pieces:
        break
        
    current = input()
    
difference = abs(number_of_pieces - counter_pieces)
if number_of_pieces > counter_pieces:
    print(f'{difference} pieces are left.')
else:
    print(f'No more cake left! You need {difference} pieces more.')
